[Enhancement] Resolve hostname in FE

imay opened the pull request:
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [x] 3.0
  - [x] 2.5

wanpengfei-git commented:
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 11 / 18 (61.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/DnsCache.java|7|14|50.00%|[32, 33, 34, 36, 37, 38, 46]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|4|4|100.00%|[]|

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

dengliu commented on fe/fe-core/src/main/java/com/starrocks/qe/DnsCache.java:
```diff
@@ -0,0 +1,48 @@
+        if (ip != null) {
+            return ip;
+        }
+        ip = InetAddress.getByName(host).getHostAddress();
```
You don't need a map here. Java, by default, has 30-second DNS cache TTL.  By the way, for those who want to explicitly check the default TTL, here is the [Code](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/sun/net/InetAddressCachePolicy.java#L48) where you can check the default TTL time.

Here is a nice article about java DNS cache:
https://medium.com/@avocadi/dns-cache-ttl-in-java-adb1c0525459

kevincai replied:
FE StarRocksMain, change it to 60 seconds. so when a pod is rebuilt with a different ip address, it may take 60s in the worst case to get the right ip address from dns server.

imay committed:
[Enhancement] Resolve hostname in frontend

Because compute node has no DNS cache in contaiener, it may cause resolving flooding for DNS server.
To avoid this scenario, we resolve hostnames in frontend. After applying this patch,
 Compute node doesn't need to resolve hostname anymore.

Fixes #32045

Signed-off-by: imay <buaa.zhaoc@gmail.com>

wanpengfei-git commented:
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/DnsCache.java|2|4|50.00%|[25, 30]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|3|3|100.00%|[]|

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
@Mergifyio backport branch-3.0

wanpengfei-git commented:
@Mergifyio backport branch-2.5
